pkgs/servers/sql/postgresql/libpq.nix

```diff
@@ -0,0 +1,61 @@
+{ stdenv
+, lib
+, openssl
+, zlib
+, libkrb5
+, icu
+, postgresql
+, pkg-config
+, tzdata
+, gssSupport ? !stdenv.hostPlatform.isWindows
+}:
+
+stdenv.mkDerivation {
+  pname = "libpq";
+  inherit (postgresql) src version;
+
+  configureFlags = [
+    "--with-openssl"
+    "--with-icu"
+    "--without-readline"
+    "--with-system-tzdata=${tzdata}/share/zoneinfo"
+    "--enable-debug"
+  ]
+  ++ lib.optionals gssSupport [ "--with-gssapi" ];
+
+  nativeBuildInputs = [ pkg-config tzdata ];
+  buildInputs = [ openssl zlib icu ]
+    ++ lib.optional gssSupport libkrb5;
+
+  enableParallelBuilding = !stdenv.isDarwin;
+
+  separateDebugInfo = true;
+
+  buildFlags = [ "submake-libpq" "submake-libpgport" ];
+
+  installPhase = ''
+    runHook preInstall
+
+    make -C src/bin/pg_config install
```

Comment:
This is wrong:
```
$ pg_config

...
LIBDIR = /nix/store/mvwrr8g7m7xk39k47qbnwsdg28cby2x2-libpq-16.1-dev/lib
...
```

This probably needs to be patched the same way as the complete postgres package. Hence it's necessary to reach an agreement with the maintainers first.